### ISSUE TITLE
migrate js-interop tests out of lib_2/html

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#40408]
Many js-interop features work without the DOM and are being used in dart2js's server mode (e.g. sass uses js-interop within node.js).

I'd like to do most of the js-interop testing in a way that doesn't require the browser to be present. Many js-interop tests are DOM independent but use dart:html for something non essential.

We'd like to migrate them, move them under `lib_2/js/`, and start running them in configurations like dart2js-d8.

**Comment by sigmundch** [comment]
@rileyporter - with your recent change, is it possible that this issue is addressed already?

**Comment by rileyporter** [comment]
Yes, I think this is addressed with my js_util test refactor. The only other lib_2/html tests that use js_util are:
- https://github.com/dart-lang/sdk/blob/master/tests/lib_2/html/js_transferrables_test.dart
- https://github.com/dart-lang/sdk/blob/master/tests/lib_2/html/js_typed_interop_dynamic_test.dart
- https://github.com/dart-lang/sdk/blob/master/tests/lib_2/html/js_typed_interop_lazy_test.dart
- https://github.com/dart-lang/sdk/blob/master/tests/lib_2/html/js_typed_interop_rename_static_test.dart

All of those tests are not testing js_util functionality directly, but are using js_util to test other functionality on window, iframe or other dart:html objects.

**Comment by sigmundch** [comment]
Nice!

I just realized that this bug was also intended to track tests that cover usage of `package:js`. It appears that, similar to what you noticed with js-util, there are a mix of tests in lib/html using it, some to cover how `package:js` works and some to use `package:js` in order to test something else.

**Comment by rileyporter** [comment]
For tracking, this merged change migrated `js_util` tests out of lib/html: https://dart-review.googlesource.com/c/sdk/+/185281

Still need to migrate other `package:js` functionality tests out of lib/html
